coffee-chats/src/main/webapp/src/App.tsx

```diff
@@ -4,11 +4,20 @@ import {MainPage} from "./components/MainPage";
 import {NavBar} from "./components/NavBar";
 import {HashRouter as Router, Switch, Route} from "react-router-dom";
 import {GroupListPage} from "./components/GroupListPage";
+import {useFetch} from "./util/fetch";
+import {AuthState} from "./entity/AuthState";
 
 function App() {
+  // this will automatically redirect to the login page if not logged in
+  const authState: AuthState | null = useFetch("/api/auth");
+
+  if (authState == null) {
```

Comment:
nit: ===

== can sometimes give surprising results re null, undefined etc in javascript. Probably it's the same in typescript

Reply:
I think Typescript fixes most of the problems with == actually